lasagne/layers/pool.py

```diff
@@ -415,8 +426,18 @@ def get_output_shape_for(self, input_shape):
     def get_output_for(self, input, **kwargs):
         a, = self.scale_factor
         upscaled = input
-        if a > 1:
-            upscaled = T.extra_ops.repeat(upscaled, a, 2)
+        if self.mode == 'repeat':
+            if a > 1:
+                upscaled = T.extra_ops.repeat(upscaled, a, 2)
+        elif self.mode == 'bed_of_nails':
+            if a > 1:
+                input_shape = input.shape
+                output_shape = self.get_output_shape_for(input_shape)
+                _, _, i = T.nonzero(input)
+                upscaled = T.zeros(shape=output_shape, dtype=input.dtype)
+                upscaled = T.set_subtensor(
+                    upscaled[:, :, i * self.scale_factor[0]],
+                    input[:, :, i])
```

Comment:
I think the `T.nonzero` will make this unnecessarily expensive. Can't you just do:
```python
output_shape = self.get_output_shape_for(input.shape)
upscaled = T.zeros(shape=output_shape, dtype=input.dtype)
upscaled = T.set_subtensor(upscaled[:, :, ::a], input)
```
Note that I'm reusing `a` for consistency with the `repeat` version.